### PR TITLE
implemented Token Based Authentication

### PR DESCRIPTION
**Pull request by Abhishek723**
Implement Token Based Authentication

### CODE REVIEW

**Review by nemish-betterhalf — COMMENTED** [review]
@Abhishek723 remove .pyc files from github. add into gitignore

File: swiggyclone/service/models.py [hunk 1]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
...
-    branch = models.ForeignKey(Branch,related_name='foodItems', on_delete=models.CASCADE)
+    branch = models.ForeignKey('service.Branch',related_name='foodItems', on_delete=models.CASCADE)
     def __str__(self):
         return self.name
 
 
 class Order(models.Model):
-    user = models.ForeignKey(User,related_name='orders', on_delete=models.CASCADE)
-    branch = models.ForeignKey(Branch,related_name='orders', on_delete=models.CASCADE)
+    user = models.ForeignKey(User,related_name='orders',null=True, blank=True, on_delete=models.CASCADE)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: why user can be null or blank?

File: swiggyclone/service/views.py [hunk 2]
```diff
@@ -22,45 +22,74 @@
...
+                    finalQuantity = instance.first().quantity - orderQuantity
+                    update_serializer = FoodItemSerializer(instance.first(), data = {"quantity":finalQuantity}, partial = True)
+                    if update_serializer.is_valid():
+                        update_serializer.save()
+                        total_price = total_price + orderQuantity*foodItemPrice
+                        
+                    else:
+                        return Response(update_serializer.errors,status = 400)
+                data["user"] = request.user
+                data["total_price"] = total_price
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: do all the stuff in serialiers's after_validation method

File: swiggyclone/accounts/serializers.py [hunk 3]
```diff
@@ -1,35 +1,7 @@
-from accounts.models import UserProfile
-from django.contrib.auth.models import User
 from rest_framework import serializers
-
-
-class UserSerializer(serializers.ModelSerializer):
+from .models import UserProfile
+class UserProfileSerializer(serializers.ModelSerializer):
+    user=serializers.StringRelatedField(read_only=True)
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/serializers.py]: why `user` is `StringRelatedField` field?

File: swiggyclone/accounts/signals.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+rom django.contrib.auth.models import User
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/signals.py]: did u test this code properly?

File: swiggyclone/accounts/views.py [hunk 5]
```diff
@@ -1,58 +1,22 @@
-from django.contrib.auth.models import User
-from django.http import JsonResponse
-from django.http import Http404
-from rest_framework import serializers
-from rest_framework.response import Response
-from rest_framework import status
-from rest_framework.views import APIView
+from rest_framework.generics import (ListCreateAPIView,RetrieveUpdateDestroyAPIView,)
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/views.py]: from rest_framework.generics import ListCreateAPIView, RetrieveUpdateDestroyAPIView

File: swiggyclone/accounts/views.py [hunk 6]
```diff
@@ -1,58 +1,22 @@
...
-from accounts.serializers import BranchOwnerRegisterSerializer,CustomerRegisterSerializer
-from accounts.models import UserProfile
-from service.models import FoodItem
-from service.serializers import FoodItemSerializer
+# Create your views here.
 
+class UserProfileListCreateView(ListCreateAPIView):
+    queryset=UserProfile.objects.all()
+    serializer_class=UserProfileSerializer
+    permission_classes=[IsAuthenticated]
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/views.py]:
```
queryset = UserProfile.objects.all()
serializer_class = UserProfileSerializer
permission_classes = [IsAuthenticated]
```

File: swiggyclone/accounts/views.py [hunk 7]
```diff
@@ -1,58 +1,22 @@
...
 
-class UserRegistertionView(APIView):
+    def perform_create(self, serializer):
+        user=self.request.user
+        serializer.save(user=user)
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/views.py]: override `create` method

File: swiggyclone/accounts/views.py [hunk 8]
```diff
@@ -1,58 +1,22 @@
...
-                return JsonResponse(serializer.data,status = 200)
-            else:
-                return JsonResponse(serializer.errors,status = 400)
-        else:
-            return Response({"message": "Permission Denied"},status = 403)
\ No newline at end of file
+class userProfileDetailView(RetrieveUpdateDestroyAPIView):
+    queryset=UserProfile.objects.all()
+    serializer_class=UserProfileSerializer
+    permission_classes=[IsOwnerProfileOrReadOnly,IsAuthenticated]
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/views.py]:
```
queryset = UserProfile.objects.all()
serializer_class = UserProfileSerializer
permission_classes = [IsOwnerProfileOrReadOnly, IsAuthenticated] 
```

Install PyCharm and install pylint to detect those type of errors.

File: swiggyclone/service/models.py [hunk 9]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
     name = models.CharField(max_length = 200)
     price = models.IntegerField()
     quantity = models.IntegerField()
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: should be positive integer field

File: swiggyclone/service/models.py [hunk 10]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
     name = models.CharField(max_length = 200)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: name = models.CharField(max_length=200)

File: swiggyclone/service/models.py [hunk 11]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
...
+    branch = models.ForeignKey('service.Branch',related_name='foodItems', on_delete=models.CASCADE)
     def __str__(self):
         return self.name
 
 
 class Order(models.Model):
-    user = models.ForeignKey(User,related_name='orders', on_delete=models.CASCADE)
-    branch = models.ForeignKey(Branch,related_name='orders', on_delete=models.CASCADE)
+    user = models.ForeignKey(User,related_name='orders',null=True, blank=True, on_delete=models.CASCADE)
+    branch = models.ForeignKey('service.Branch',related_name='orders', on_delete=models.CASCADE)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: branch = models.ForeignKey(Branch, related_name='orders', on_delete=models.CASCADE)

File: swiggyclone/service/models.py [hunk 12]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
...
     order_time = models.DateTimeField(default=timezone.now, null=True, blank=True)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: why order_time can be null or blank? is it good design?

File: swiggyclone/service/models.py [hunk 13]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
...
     total_price = models.IntegerField(default=0)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: should be a positive integer field

File: swiggyclone/accounts/signals.py (hunk 4 above)
Comment (Abhishek723) [reply swiggyclone/accounts/signals.py]: Yes I have tested it. Its working fine.
https://lewiskori.com/blog/user-registration-and-authorization-on-a-django-api-with-djoser-and-json-web-tokens/

File: swiggyclone/service/models.py [hunk 14]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
...
 
 
 class OrderDiscription(models.Model):
-    order = models.ForeignKey(Order,related_name='orderDiscriptions', on_delete=models.CASCADE)
-    foodItem = models.ForeignKey(FoodItem,related_name='orderDiscriptions', on_delete=models.CASCADE)
+    order = models.ForeignKey('service.Order',related_name='orderDiscriptions', on_delete=models.CASCADE)
+    foodItem = models.ForeignKey('service.FoodItem',related_name='orderDiscriptions', on_delete=models.CASCADE)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]:
```
order = models.ForeignKey(Order, related_name='orderDiscriptions', on_delete=models.CASCADE)
foodItem = models.ForeignKey(FoodItem, related_name='orderDiscriptions', on_delete=models.CASCADE)
```

File: swiggyclone/service/models.py [hunk 15]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
...
     quantity = models.IntegerField()
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: positive integer field

File: swiggyclone/service/models.py [hunk 16]
```diff
@@ -25,20 +25,20 @@ class FoodItem(models.Model):
...
     price = models.IntegerField()
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: positive integer field

File: swiggyclone/accounts/signals.py (hunk 4 above)
Comment (nemish-betterhalf) [reply swiggyclone/accounts/signals.py]: is this correct import statement? it should throw error. test again bro.

File: swiggyclone/accounts/views.py (hunk 5 above)
Comment (Abhishek723) [reply swiggyclone/accounts/views.py]: Okk

File: swiggyclone/service/serializers.py [hunk 17]
```diff
@@ -12,25 +12,17 @@ class OrderDiscriptionSerializers(serializers.ModelSerializer):
...
         model = OrderDiscription
-        feilds = (
+        fields = (
                 'id',
                 'price',
                 'quantity', 
-                'order_id',
-                'foodItem_id'  
+                'foodItem'  
             )
```
Comment (nemish-betterhalf) [inline swiggyclone/service/serializers.py]: fields = ( 'id',  'price', ...)
 put atmost 120 character in single line

File: swiggyclone/service/serializers.py [hunk 18]
```diff
@@ -39,7 +31,7 @@ class Meta:
                   'price',
                   'quantity',
                   'orderDiscriptions',
-                  'branch_id'
```
Comment (nemish-betterhalf) [inline swiggyclone/service/serializers.py:42]: fields = ( 'price', ...)
put atmost 120 character in single line

File: swiggyclone/service/serializers.py [hunk 19]
```diff
@@ -48,13 +40,12 @@ class OrderSerializers(serializers.ModelSerializer):
...
-        feilds = (
+        fields = (
                 'id',
                 'order_time',
                 'total_price',
                 'orderDiscriptions',
-                'user_id',
-                'branch_id'
+                'branch'
             )
```
Comment (nemish-betterhalf) [inline swiggyclone/service/serializers.py]: fields = ( 'id', 'order_time', ...)
put atmost 120 character in single line

File: swiggyclone/service/serializers.py [hunk 20]
```diff
@@ -68,49 +59,32 @@ def create(self, validated_data):
...
                   'name',
                   'address',
-                  'pincode',
-                  'foodItems',
-                  'orders',
-                  'restaurent_id',
-                  'branchOwner_id'
+                  'pincode',    
+                  'branchOwner',
+                  'restaurent'
```
Comment (nemish-betterhalf) [inline swiggyclone/service/serializers.py]: new not write each field in new line

File: swiggyclone/service/serializers.py [hunk 21]
```diff
@@ -68,49 +59,32 @@ def create(self, validated_data):
...
+    
+   
     class Meta:
         model = Restaurent
         fields = (
                   'id',
                   'name',
                   'discription',
-                  'branches',
                 )
```
Comment (nemish-betterhalf) [inline swiggyclone/service/serializers.py]: same as above

File: swiggyclone/service/serializers.py [hunk 22]
```diff
@@ -68,49 +59,32 @@ def create(self, validated_data):
...
-    def create(self, validated_data):
-        branches_data = validated_data.pop('branches')
-        restaurent = Restaurent.objects.create(**validated_data)
-        for branch_data in branches_data:
-            Branch.objects.create(restaurent=restaurent, **branch_data)
-        return restaurent
+
```
Comment (nemish-betterhalf) [inline swiggyclone/service/serializers.py]: remove the unnecessary lines at the end of files. there should be one empty line at the end of the file. not less or more

File: swiggyclone/service/urls.py [hunk 23]
```diff
@@ -1,9 +1,5 @@
 from django.urls import path,include
 from service.views import (
-    PlaceOrder,
-    RestaurentRegistertionView,
-    BranchRegistertionView,
-    FoodItemRegistertionView,
     RestaurentViewSet,
     BranchViewSet,
     FoodItemViewSet
```
Comment (nemish-betterhalf) [inline swiggyclone/service/urls.py]: do it in single line and remove ( and ) at all places if not required

File: swiggyclone/service/views.py [hunk 24]
```diff
@@ -22,45 +22,74 @@
 
 class RestaurentViewSet(viewsets.ModelViewSet):
     queryset = Restaurent.objects.all().prefetch_related('branches')
-    serializer_class = RestaurentSerializers
+    serializer_class = RestaurentSerializer
 
 
 class BranchViewSet(viewsets.ModelViewSet):
     queryset = Branch.objects.all().prefetch_related('foodItems')
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: you have defined `queryset` variable as well as `get_queryset()` method. have you checked both is required or only one is enough in your case?

File: swiggyclone/service/views.py [hunk 25]
```diff
@@ -22,45 +22,74 @@
...
-                if serializer.is_valid():
-                    serializer.save()
-                    total_price = total_price + orderQuantity*foodItemPrice
+                try: 
+                    instance = FoodItem.objects.filter(branch_id=orderedBranch, id=orderFoodItem)
+                except:
+                    orderValid = False
+                if orderValid:
+                    serialize = FoodItemSerializer(instance.first())
+                    foodItemQuantity = serialize.data["quantity"]
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: do proper use of serializer.

File: swiggyclone/service/views.py [hunk 26]
```diff
@@ -22,45 +22,74 @@
...
-                    return Response(serializer.errors,status = 400)
+                    orderValid = False
+                    return Response(orderSerializers.errors,status = 404)
         else:
-            Response(serializer.errors,status = 404)
+            orderValid = False
+            return Response(orderSerializers.errors,status = 404)
+        if orderValid:
+            return Response({"message":"Order Placed"}, status = 200)
+        return Response({"message":"Food Item is not present"},status = 404)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: if possible check those conditions in serializer and raise errors from serializer. raise ValidationError('...')

File: swiggyclone/service/views.py [hunk 27]
```diff
@@ -22,45 +22,74 @@
...
-        return Response({"message":"Order Placed", "total Price":total_price}, status = 200)
 
 
 class FoodItemViewSet(viewsets.ModelViewSet):
     queryset = FoodItem.objects.all()
     serializer_class = FoodItemSerializer
     def get_queryset(self):
-        return FoodItem.objects.filter(restaurent_id = self.kwargs['restaurent_pk'], branch_id = self.kwargs['branch_pk'])
\ No newline at end of file
+        return FoodItem.objects.filter(branch_id = self.kwargs['branch_pk'])
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: have checked that both required `queryset` and `get_queryset`?

File: swiggyclone/accounts/serializers.py (hunk 3 above)
Comment (nemish-betterhalf) [reply swiggyclone/accounts/serializers.py]: user = seriai.....

File: swiggyclone/service/models.py [hunk 28]
```diff
@@ -14,7 +14,7 @@ def __str__(self):
 class Branch(models.Model):
     name = models.CharField(max_length = 200) 
     address = models.TextField(max_length=500)
-    restaurent = models.ForeignKey(Restaurent,related_name='branches', on_delete=models.CASCADE)
+    restaurent = models.ForeignKey('service.Restaurent',related_name='branches', on_delete=models.CASCADE)
     branchOwner = models.ForeignKey(User,related_name='branches', on_delete=models.CASCADE)
     pincode = models.CharField(max_length = 6)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/models.py]: remove whitespace

File: swiggyclone/accounts/views.py [hunk 29]
```diff
@@ -10,9 +11,9 @@ class UserProfileListCreateView(ListCreateAPIView):
     serializer_class = UserProfileSerializer
     permission_classes = [IsAuthenticated]
 
-    def create(self, validated_data):
-        validated_data['user'] = self.request.user
-        return super(UserProfileListCreateView, self).create(validated_data)
+    def perform_create(self, validated_data):
+        user = self.request.user
+        serializer.save(user=user)
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/views.py]: why?

File: swiggyclone/service/views.py [hunk 30]
```diff
@@ -11,67 +12,34 @@ class RestaurentViewSet(viewsets.ModelViewSet):
...
-                        return Response(update_serializer.errors, status=400)
-                data["user"] = request.user
-                data["total_price"] = total_price
-                finalOrderSerializer = OrderSerializers(data=data)
-                if finalOrderSerializer.is_valid():   
-                    finalOrderSerializer.save()
-                else:
-                    orderValid = False
-                    return Response(orderSerializers.errors, status=404)
+        data['branch'] = self.kwargs['pk']
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py:38]: do not pass in data. use context

File: swiggyclone/service/views.py [hunk 31]
```diff
@@ -11,67 +12,34 @@ class RestaurentViewSet(viewsets.ModelViewSet):
...
+        serializer = OrderSerializers(data=data)
+        if serializer.is_valid():
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: serializer.is_valid(raise_exception=True)
serializer.save()

No need to write else condition

File: swiggyclone/service/views.py [hunk 32]
```diff
@@ -11,67 +12,34 @@ class RestaurentViewSet(viewsets.ModelViewSet):
...
+            serializer.save()
+            return Response("Order Placed", status=status.HTTP_200_OK)
         else:
-            orderValid = False
-            return Response(orderSerializers.errors, status=404)
-        if orderValid:
-            return Response({"message": "Order Placed"}, status=200)
-        return Response({"message": "Food Item is not present"}, status=404)
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: not required

File: swiggyclone/service/views.py [hunk 33]
```diff
@@ -11,67 +12,34 @@ class RestaurentViewSet(viewsets.ModelViewSet):
...
 
 
 class FoodItemViewSet(viewsets.ModelViewSet):
     serializer_class = FoodItemSerializer
 
     def get_queryset(self):
         return FoodItem.objects.filter(branch_id=self.kwargs['branch_pk'])
+
+    def perform_create(self, serializer):
+        serializer.save(branch_id=self.kwargs['branch_pk'])
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: what is this?

File: swiggyclone/service/views.py [hunk 34]
```diff
@@ -11,67 +12,34 @@ class RestaurentViewSet(viewsets.ModelViewSet):
...
-            return Response({"message": "Order Placed"}, status=200)
-        return Response({"message": "Food Item is not present"}, status=404)
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
 class FoodItemViewSet(viewsets.ModelViewSet):
     serializer_class = FoodItemSerializer
 
     def get_queryset(self):
         return FoodItem.objects.filter(branch_id=self.kwargs['branch_pk'])
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: what if i will pass `branch_pk="abcded"`?

File: swiggyclone/service/views.py (hunk 30 above)
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py:38]: what if i will pass `pk='fyfb76jbdnjsd'`?

File: swiggyclone/service/views.py [hunk 35]
```diff
@@ -11,67 +12,34 @@ class RestaurentViewSet(viewsets.ModelViewSet):
     queryset = Restaurent.objects.all().prefetch_related('branches')
     serializer_class = RestaurentSerializer
 
+
 class BranchViewSet(viewsets.ModelViewSet):
     serializer_class = BranchSerializer
+
     def get_queryset(self):
         return Branch.objects.filter(restaurent_id=self.kwargs['restaurent_pk'])
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: what happens if i will pass `restaurent_pk='ahsb41515'`? have u tested?

File: swiggyclone/service/serializers.py [hunk 36]
```diff
@@ -1,41 +1,68 @@
...
+        fields = ['id', 'order_time', 'orderDiscriptions', 'branch', 'user']
+
+    def validate(self, data):
+        orderDiscriptions_data = data['orderDiscriptions']
+        for orderDiscription_data in orderDiscriptions_data:
+            orderFoodItem = orderDiscription_data['foodItem']
+            orderQuantity = orderDiscription_data['quantity']
+            foodItemQuantity = orderDiscription_data['foodItem'].quantity
+            if foodItemQuantity < orderQuantity:
+                raise serializers.ValidationError('Given fooditem not found.')
```
Comment (nemish-betterhalf) [inline swiggyclone/service/serializers.py:48]: `Invalid fooditem`

File: .idea/Clone.iml [hunk 37]
```diff
@@ -0,0 +1,15 @@
...
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+  <component name="PyDocumentationSettings">
+    <option name="format" value="PLAIN" />
+    <option name="myDocStringFormat" value="Plain" />
+  </component>
+  <component name="TestRunnerService">
+    <option name="PROJECT_TEST_RUNNER" value="pytest" />
+  </component>
```
Comment (nemish-betterhalf) [inline .idea/Clone.iml]: remove these files

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 38]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <settings>
+    <option name="USE_PROJECT_PROFILE" value="false" />
+    <version value="1.0" />
+  </settings>
+</component>
```
Comment (nemish-betterhalf) [inline .idea/inspectionProfiles/profiles_settings.xml]: remove these files

File: .idea/misc.xml [hunk 39]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.8" project-jdk-type="Python SDK" />
+  <component name="PyCharmProfessionalAdvertiser">
+    <option name="shown" value="true" />
+  </component>
```
Comment (nemish-betterhalf) [inline .idea/misc.xml]: remove these files

File: .idea/vcs.xml [hunk 40]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (nemish-betterhalf) [inline .idea/vcs.xml]: remove these files

File: .idea/modules.xml [hunk 41]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/Clone.iml" filepath="$PROJECT_DIR$/.idea/Clone.iml" />
+    </modules>
+  </component>
+</project>
```
Comment (nemish-betterhalf) [inline .idea/modules.xml]: remove these files

**Review by nemish-betterhalf — CHANGES_REQUESTED** [review]
@Abhishek723 remove all .idea and pycharm related files from github. you can add into .gitignores

**Review by nemish-betterhalf — COMMENTED** [review]
@Abhishek723 where are your migrations files? please add those migrations file to Github.

File: swiggyclone/accounts/models.py [hunk 42]
```diff
@@ -1,15 +1,21 @@
...
+    user = models.OneToOneField(User, on_delete=models.CASCADE, related_name="profile")
+    address = models.CharField(max_length=500, blank=True)
+    mobile_number = models.CharField(max_length=10, blank=True)
     is_branchOwner = models.BooleanField(default=False)
-    
\ No newline at end of file
+    
+    def __str__(self):
+        return self.user.username
+@receiver(post_save, sender=User)
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/models.py]: put 2 empty lines here

File: swiggyclone/accounts/models.py [hunk 43]
```diff
@@ -1,15 +1,21 @@
...
+def create_user_profile(sender, instance, created, **kwargs):
+    if created:
+        UserProfile.objects.create(user=instance)
+
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/models.py:20]: put one more empty line here

File: swiggyclone/accounts/models.py [hunk 44]
```diff
@@ -1,15 +1,21 @@
...
+@receiver(post_save, sender=User)
+def save_user_profile(sender, instance, **kwargs):
+    instance.profile.save()
```
Comment (nemish-betterhalf) [inline swiggyclone/accounts/models.py]: there should be one empty line at the end of each file

**Pull request merged**